web-ui/src/main/resources/catalog/components/edit/onlinesrc/OnlineSrcDirective.js

```diff
@@ -818,39 +818,53 @@
                           })
                         }));
                       });
-
-                  $timeout(function() {
-                    if (angular.isArray(scope.gnCurrentEdit.extent)) {
-                      // FIXME : only first extent is took into account
-                      var projectedExtent;
-                      var extent = scope.gnCurrentEdit.extent &&
-                          scope.gnCurrentEdit.extent[0];
-                      var proj = ol.proj.get(gnMap.getMapConfig().projection);
-
-                      if (!extent || !ol.extent.containsExtent(
-                          proj.getWorldExtent(),
-                          extent)) {
-                        projectedExtent = proj.getExtent();
-                      }
-                      else {
-                        projectedExtent =
-                            gnMap.reprojExtent(extent, 'EPSG:4326', proj);
-                      }
-                      scope.map.getView().fit(
-                          projectedExtent,
-                          scope.map.getSize());
-                    }
-                    // Trigger init of print directive
-                    scope.mode = 'thumbnailMaker';
-                  }, 300);
+                  
```

Comment:
Refactor, not change: a more angular way that not only do a timeout and cross fingers to see if the map is there, but really wait for the map to be there.